dm/master/server.go

```diff
@@ -170,11 +170,8 @@ func (s *Server) Start(ctx context.Context) (err error) {
 	if err != nil {
 		return terror.ErrMasterTLSConfigNotValid.Delegate(err)
 	}
-	if tls2 != nil && tls2.TLSConfig() != nil {
-		tls2.TLSConfig().InsecureSkipVerify = true
-	}
 
-	apiHandler, err := getHTTPAPIHandler(ctx, s.cfg.MasterAddr, tls2.ToGRPCDialOption())
```

Comment:
Sometimes `MasterAddr` is not an IP address tls will return error `transport: authentication handshake failed: tls: either ServerName or InsecureSkipVerify must be specified in the tls.Config`. And `AdvertiseAddr`'s format looks like `127.0.0.1:8261`, and tls will set `ServerName` to `127.0.0.1` here https://golang.org/src/crypto/tls/tls.go?h=ServerName#L162

Reply:
`MasterAddr` should be the bind address of the dm-master, it should be an IP address, right?

`AdvertiseAddr` can be a domain.

Reply:
what's the required format of `MasterAddr` and `AdvertiseAddr`, might `adjust` them in advance. Currently we will set `AdvertiseAddr` to `MasterAddr` if empty

Reply:
Seems we should also set `tls.Config.ServerName` to its domain, (though I'm not sure which address of config this domain should came from)